llvm/include/llvm/SYCLLowerIR/DeviceConfigFile.td

```diff
@@ -0,0 +1,108 @@
+    AspectUsm_atomic_shared_allocations];
+
+
+def TargetTable : DynamicTable {
+  let FilterClass = "TargetInfo";
+  let Fields = ["DeviceId", "maySupportOtherAspects", "aspects",
+                "subGroupSizes", "aotToolchain", "aotToolchainOptions"];
+  string TypeOf_aspects = "list<Aspect>";
+  string TypeOf_subGroupSizes = "list<int>";
+}
+
+class TargetInfo<string targetName, list<Aspect> aspectList, list<int> subGroupSizesList, string toolchain = "", string options = "", bit maySupportOtherAspects_init = 0> {
+  string DeviceId = targetName;
+  bits<1> maySupportOtherAspects = maySupportOtherAspects_init;
+  list<Aspect> aspects = aspectList;
+  list<int> subGroupSizes = subGroupSizesList;
+  string aotToolchain = toolchain;
+  string aotToolchainOptions = options;
+}
+
+// This definition serves the only purpose of testing whether the aspect list defined in here and in SYCL RT match.
+def : TargetInfo<"__TestAspectList",
+    [AspectCpu, AspectGpu, AspectAccelerator, AspectCustom, AspectFp16, AspectFp64, AspectImage, AspectOnline_compiler, 
+    AspectOnline_linker, AspectQueue_profiling, AspectUsm_device_allocations, AspectUsm_host_allocations,
+    AspectUsm_shared_allocations, AspectUsm_system_allocations, AspectExt_intel_pci_address,
+    AspectExt_intel_gpu_eu_count, AspectExt_intel_gpu_eu_simd_width, AspectExt_intel_gpu_slices,
+    AspectExt_intel_gpu_subslices_per_slice, AspectExt_intel_gpu_eu_count_per_subslice,
+    AspectExt_intel_max_mem_bandwidth, AspectExt_intel_mem_channel, AspectUsm_atomic_host_allocations,
+    AspectUsm_atomic_shared_allocations, AspectAtomic64, AspectExt_intel_device_info_uuid, AspectExt_oneapi_srgb,
+    AspectExt_oneapi_native_assert, AspectHost_debuggable, AspectExt_intel_gpu_hw_threads_per_eu,
+    AspectExt_oneapi_cuda_async_barrier, AspectExt_oneapi_bfloat16_math_functions, AspectExt_intel_free_memory,
+    AspectExt_intel_device_id, AspectExt_intel_memory_clock_rate, AspectExt_intel_memory_bus_width, AspectEmulated,
+    AspectExt_intel_legacy_image],
+    []>;
+// This definition serves the only purpose of testing whether the deprecated aspect list defined in here and in SYCL RT
+// match.
+def : TargetInfo<"__TestDeprecatedAspectList",
+    [AspectInt64_base_atomics, AspectInt64_extended_atomics, AspectUsm_system_allocator,
+    AspectUsm_restricted_shared_allocations, AspectHost],
+    []>;
+
+def : TargetInfo<"TargetA", [AspectCpu, AspectAtomic64], [8, 16], "ocloc", "devA", 1>;
+def : TargetInfo<"TargetB", [AspectGpu, AspectFp16], [8, 16], "ocloc", "devB">;
+def : TargetInfo<"TargetC", [AspectEmulated, AspectImage], [8, 16], "ocloc", "devC">;
+def : TargetInfo<"TargetD", [AspectCpu, AspectFp64, AspectHost_debuggable], [8, 16], "ocloc", "devD", 1>;
+defvar AspectList = [AspectCpu] # AllUSMAspects;
+def : TargetInfo<"Test", AspectList, []>;
+def : TargetInfo<"Test2", [AspectCpu] # AllUSMAspects, []>;
```

Comment:
I understand that those are examples, but we should either remove them or move to comments before the commit.

I think it makes sense to leave one simple example of adding a new target here and one example, where list of aspects is produced as a combination of explicitly specified values + some other predefined list.